espnet2/s2st/synthesizer/discrete_synthesizer.py

```diff
@@ -256,13 +254,13 @@ def forward_one_step(
         # FIXME(jiatong): the spk/lang embedding may be execute too many times
         # consider add before the search
         if self.spks is not None:
-            sid_embs = self.sid_emb(sids.view(-1))
+            sid_embs = self.sid_emb(self.spks.view(-1))
             memory = memory + sid_embs.unsqueeze(1)
         if self.langs is not None:
-            lid_embs = self.lid_emb(lids.view(-1))
+            lid_embs = self.lid_emb(self.langs.view(-1))
             memory = memory + lid_embs.unsqueeze(1)
         if self.spk_embed_dim is not None:
-            memory = self._integrate_with_spk_embed(memory, spembs)
+            memory = self._integrate_with_spk_embed(memory, self.spk_embed_dim)
```

Comment:
Thanks for your check and update. Upon double-checking the file with speaker ID, language ID, and Speaker embedding, I think this part is actually not needed (actually this part of the code is not used in the current implementation at all). 

However, since this part is not a focus of this PR, I will start a new PR specifically with this fix very soon. In that case, how about keeping it as it is for this PR?